static/js/browser2/observation_chart.tsx

```diff
@@ -81,159 +76,61 @@ export class ObservationChart extends React.Component<
 
   private plot(): void {
     const values = this.props.sourceSeries.val;
-    if (_.isEmpty(values)) {
-      return;
-    }
-    const plotData = [];
-    Object.keys(values).forEach((key) =>
-      plotData.push({
-        dateString: key,
-        time: Date.parse(key),
+    const data = [];
+    Object.keys(values).forEach((key) => {
+      data.push({
+        label: key,
         value: Number(values[key]),
-      })
+      });
+    });
+    const dataGroups = [new DataGroup("", data)];
+    const svgContainerId: string = "svg-container" + this.props.idx;
+    drawLineChart(
+      svgContainerId,
+      WIDTH,
+      HEIGHT,
+      dataGroups,
+      true,
+      this.getUnits(),
+      this.handleDotClick
     );
-
-    const svg = d3
-      .select("#svg-container" + this.props.idx)
-      .append("svg")
-      .attr("width", WIDTH + LEFT + RIGHT)
-      .attr("height", HEIGHT + TOP + BOTTOM)
-      .append("g")
-      .attr("transform", `translate(${LEFT},${TOP})`);
-
-    // Set the ranges
-    const x = d3.scaleTime().range([0, WIDTH]);
-    const y = d3.scaleLinear().range([HEIGHT, 0]);
-
-    x.domain(d3.extent(plotData, (d) => d.time));
-
-    const yMax = d3.max(plotData, (d) => d.value);
-    const yMin = d3.min(plotData, (d) => d.value);
-
-    // When yMax and yMin are almost equal, need have a non-zero range for d3 to
-    // plot the y axis.
-    // If the min value is 0, then always start from it. Otherwise choose a tick
-    // range that covers yMin and yMax.
-    if (yMax - yMin < 0.0001) {
```

Comment:
I'm not sure of exact nodes where these problems manifested, but it would be good to see if these are fixed by the shared charting code.

Reply:
Done